src/main/target/common_post.h

```diff
@@ -552,14 +552,43 @@
 #undef USED_TIMERS
 #endif
 
-#if !defined(USE_RANGEFINDER)
+#ifdef USE_RANGEFINDER
```

Comment:
This means that the is no way to select only ane rangefinder type (neither in target.h, noor on command line)

Reply:
It is the same we do with USE_MAG at the moment. Then should we turn it around if one of the rangefinders is included in the build we also add USE_RANGFINDER ?

Reply:
Maybe just undef all when instances when USE_RANGFINDER is not defined .. 
Some better mechanism is needed, until then try keeping status quo.

I can imagine using TOML or similar as build option database and then generating _post.h automatically (look at serial normalization, python+jinja is quite easy to follow)

With auto-generated normalization, you can use different defines prefix for  target/platform/commandline + use values (enable, disable, default,...).
Output will be classic #USE_ , so no changes to C code

Reply:
Definitely need a better mechanism. Post is a mess. Flash is a good example of this problem.

Given it's not a space issue, just load all hardware if rangefinder is added, and no other specific hardware item is present.

That way it works in config.h for specific on board hardware as well as for users who specify hardware and range finder or just range finder.